lms/djangoapps/email_marketing/signals.py

```diff
@@ -25,6 +27,28 @@
                       'age', 'level_of_education', 'year_of_birth',
                       'country', LANGUAGE_KEY]
 
+WAFFLE_NAMESPACE = 'sailthru'
+WAFFLE_SWITCHES = WaffleSwitchNamespace(name=WAFFLE_NAMESPACE)
+
+SAILTHRU_AUDIT_PURCHASE_ENABLED = 'audit_purchase_enabled'
+
+
+@receiver(ENROLL_STATUS_CHANGE)
+def update_sailthru(sender, event, user, mode, course_id, **kwargs):
+    """
+    Receives signal and calls a celery task to update the
+    enrollment track
+    Arguments:
+        user: current user
+        course_id: course key of a course
+    Returns:
+        None
+    """
+    if WAFFLE_SWITCHES.is_enabled(SAILTHRU_AUDIT_PURCHASE_ENABLED) and mode != 'verified':
```

Comment:
Checking on mode==`verified` is not enough.
You should check all modes with price. 
Maybe it's easier make the condition as `mode in ('audit', 'honor')`. By the way, please don't use strings. Use the constants defined at https://github.com/edx/edx-platform/blob/master/common/djangoapps/course_modes/models.py#L159